### PR TITLE
Improve performance of hideRegex in logging

### PR DESCRIPTION
**Pull request by tnleeuw**

### CODE REVIEW

File: commons/src/main/java/org/frankframework/util/StringUtil.java [hunk 1]
```diff
@@ -157,9 +156,17 @@ public static String hideAll(String inputString, String regex) {
...
 	public static String hideAll(String inputString, String regex, int mode) {
+		return hideAll(inputString, Pattern.compile(regex), mode);
+	}
+
+	/**
+	 * Hides the input string according to the given regex and mode.
+	 * If mode is set to 1, then the first half of the string gets hidden.
+	 * Else, all of it.
+	 */
+	public static String hideAll(String inputString, Pattern regex, int mode) {
```
Comment (jkosternl) [inline commons/src/main/java/org/frankframework/util/StringUtil.java:167]: We still need both (overloaded) methods?

**Review by jkosternl — APPROVED** [review]

File: core/src/main/java/org/frankframework/logging/IbisMaskingLayout.java [hunk 2]
```diff
@@ -58,12 +58,12 @@ public abstract class IbisMaskingLayout extends AbstractStringLayout {
 	 * Set of regex strings to hide locally, meaning for specific threads/adapters.
 	 * This is required to be set for each thread individually.
 	 */
-	private static final ThreadLocal<Set<String>> threadLocalReplace = new ThreadLocal<>();
+	private static final ThreadLocal<Map<String, Pattern>> threadLocalReplace = new ThreadLocal<>();
```
Comment (evandongen) [inline core/src/main/java/org/frankframework/logging/IbisMaskingLayout.java:61]: Perhaps this is too much for now, but why not refactor this to a Set<Pattern>? That way you can only access the compiled Patterns via the ThreadLocal

File: commons/src/main/java/org/frankframework/util/StringUtil.java (hunk 1 above)
Comment (tnleeuw) [reply commons/src/main/java/org/frankframework/util/StringUtil.java:167]: For now, there are still callers that pass a string where I didn't feel it necessary to optimize.
That can perhaps come in a future PR, if the overloaded version with a `String regex` is no longer used.

File: core/src/main/java/org/frankframework/logging/IbisMaskingLayout.java (hunk 2 above)
Comment (tnleeuw) [reply core/src/main/java/org/frankframework/logging/IbisMaskingLayout.java:61]: Let me see -- I thought that under some circumstances, we still wanted to be able to look at the original regexes or be able to remove by original regex. Although that method is perhaps never actually used.

**Review by evandongen — APPROVED** [review]

File: core/src/main/java/org/frankframework/logging/IbisMaskingLayout.java (hunk 2 above)
Comment (tnleeuw) [reply core/src/main/java/org/frankframework/logging/IbisMaskingLayout.java:61]: I looked at it, and I think this is a good suggestion. I will update this.

**Review by nielsm5 — APPROVED** [review]

**Pull request merged**